specifyweb/frontend/js_src/lib/components/Interactions/InteractionDialog.tsx

```diff
@@ -288,12 +288,22 @@ export function InteractionDialog({
                 >
                   {interactionsText.addUnassociated()}
                 </Button.Info>
-              ) : model.name === 'Loan' ||
-                (action.model.name === 'Loan' && prepsData?.length === 0) ? (
+              ) : (model.name === 'Loan' || action.model.name === 'Loan') &&
+                !(
+                  state.type === 'MissingState' &&
+                  prepsData?.length !== 0 &&
+                  prepsData
+                ) ? (
                 <Link.Blue href={getResourceViewUrl('Loan')}>
                   {interactionsText.withoutPreparations()}
                 </Link.Blue>
-              ) : undefined}
+              ) : null}
+              {model.name === 'Gift' ||
+              (action.model.name === 'Gift' && itemCollection === undefined) ? (
```

Comment:
sorry I didn't read the surrounding code much, but it looks suspicious that the condition for Loan button and Gift button look so different

Also, can you please look at the code in version `v7.9.0` or before to see how this looked like before the #4076 bug?
If the task is to bring back a button that was accidentally removed, then the code that was here before the change would be a good guide

Reply:
it didn't exist before. We only had a button for loans